File: samples/SelfAspNet/SelfAspNet/Chap10/MyApp.Master.cs
using System;
using System.Data;
using System.Web.UI;

namespace SelfAspNet.Chap10
{
    public partial class MyApp : System.Web.UI.MasterPage
    {
        protected void Page_Load(object sender, EventArgs e)
        {
            sds.SelectParameters.Clear();
            sds.SelectParameters.Add("url", Request.AppRelativeCurrentExecutionFilePath);
            var reader = (IDataReader)sds.Select(DataSourceSelectArguments.Empty);
            if (reader.Read())
            {
                Page.Header.Title = reader["title"].ToString();
                Page.Header.Keywords = reader["keywd"].ToString();
                Page.Header.Description = reader["description"].ToString();
            }
        }
    }
}

File: samples/SelfAspNetVb/SelfAspNetVb/Chap10/MyApp.Master.vb
Public Class MyApp
    Inherits System.Web.UI.MasterPage

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        sds.SelectParameters.Clear()
        sds.SelectParameters.Add("url", Request.AppRelativeCurrentExecutionFilePath)
        Dim reader As IDataReader = sds.Select(DataSourceSelectArguments.Empty)
        If reader.Read() Then
            Page.Header.Title = reader("title")
            Page.Header.Keywords = reader("keywd")
            Page.Header.Description = reader("description")
        End If
    End Sub

End Class